projects/com.oracle.truffle.llvm.parser.impl/src/com/oracle/truffle/llvm/parser/impl/lifetime/LLVMLifeTimeAnalysisVisitor.java

```diff
@@ -1,392 +0,0 @@
- * with the distribution.
- *
- * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
- * endorse or promote products derived from this software without specific prior written
- * permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
- * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
- * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
- * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
- * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
- * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
- * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
- * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
- * OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-package com.oracle.truffle.llvm.parser.impl.lifetime;
-
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.stream.Collectors;
-
-import org.eclipse.emf.common.util.EList;
-import org.eclipse.emf.ecore.EObject;
-
-import com.intel.llvm.ireditor.lLVM_IR.BasicBlock;
-import com.intel.llvm.ireditor.lLVM_IR.FunctionDef;
-import com.intel.llvm.ireditor.lLVM_IR.Instruction;
-import com.intel.llvm.ireditor.lLVM_IR.Instruction_br;
-import com.intel.llvm.ireditor.lLVM_IR.Instruction_indirectbr;
-import com.intel.llvm.ireditor.lLVM_IR.Instruction_ret;
-import com.intel.llvm.ireditor.lLVM_IR.Instruction_switch;
-import com.intel.llvm.ireditor.lLVM_IR.Instruction_unreachable;
-import com.intel.llvm.ireditor.lLVM_IR.MiddleInstruction;
-import com.intel.llvm.ireditor.lLVM_IR.NamedMiddleInstruction;
-import com.intel.llvm.ireditor.lLVM_IR.NamedTerminatorInstruction;
-import com.intel.llvm.ireditor.lLVM_IR.StartingInstruction;
-import com.intel.llvm.ireditor.lLVM_IR.TerminatorInstruction;
-import com.intel.llvm.ireditor.lLVM_IR.impl.Instruction_brImpl;
-import com.intel.llvm.ireditor.lLVM_IR.impl.LocalValueRefImpl;
-import com.oracle.truffle.api.frame.FrameDescriptor;
-import com.oracle.truffle.api.frame.FrameSlot;
-import com.oracle.truffle.llvm.parser.base.util.LLVMParserAsserts;
-import com.oracle.truffle.llvm.parser.impl.LLVMPhiVisitor;
-import com.oracle.truffle.llvm.parser.impl.LLVMPhiVisitor.Phi;
-import com.oracle.truffle.llvm.parser.impl.LLVMReadVisitor;
-import com.oracle.truffle.llvm.runtime.options.LLVMOptions;
-
-/**
- * This class determines which variables are dead after each basic block. It applies an iterative
- * data-flow analysis to determine the lifetimes.
- *
- */
-public final class LLVMLifeTimeAnalysisVisitor {
```

Comment:
We cannot remove this class yet, since it is essential for peak performance (when using the textual parser). Only when we remove the textual parser completely, we can also remove it. This is also why the gate fails with a Graal bailout.